### ISSUE TITLE
Can't get the plugin to load texture packs on Mupen64plus

### ISSUE THREAD

**Issue by Redyn** [gonetz/GLideN64#1840]
All right, so I'm trying play Banjo-Kazooie on Mupen64plus-GUI with a texture pack I downloaded. It was originally in the form of a folder containing the textures, so I used Project64 with GLideN64 to convert it to .htc. After this, I moved Banjo-Kazooie_HIRESTEXTURES.htc to D:\Emulators\mupen64plus\cache. I run BK and nothing, the textures are the default ones. I tried moving the folder with the textures to D:\Emulators\mupen64plus\hires_texture as well and still nothing. What am I doing wrong? The plugin can load them just fine when I'm using P64.

My GLideN64 settings are the following:

configVersion = 21
CropMode = 0
CropWidth = 0
CropHeight = 0
MultiSampling = 0
AspectRatio = 1
BufferSwapMode = 0
UseNativeResolutionFactor = 0
bilinearMode = True
MaxAnisotropy = False
EnableNoise = True
EnableLOD = True
EnableHWLighting = False
EnableShadersStorage = True
CorrectTexrectCoords = 0
EnableNativeResTexrects = False
EnableLegacyBlending = False
EnableFragmentDepthWrite = True
EnableCustomSettings = True
EnableFBEmulation = True
EnableCopyAuxiliaryToRDRAM = False
EnableN64DepthCompare = False
ForceDepthBufferClear = False
DisableFBInfo = True
FBInfoReadColorChunk = False
FBInfoReadDepthChunk = True
EnableCopyColorToRDRAM = 2
EnableCopyDepthToRDRAM = 2
EnableCopyColorFromRDRAM = False
txFilterMode = 0
txEnhancementMode = 0
txDeposterize = False
txFilterIgnoreBG = False
txCacheSize = 3000
txHiresEnable = True
txHiresFullAlphaChannel = True
txHresAltCRC = False
txDump = False
txCacheCompression = True
txForce16bpp = False
txSaveCache = True
txPath = "D:\Emulators\mupen64plus\hires_texture"
txCachePath = "D:\Emulators\mupen64plus\cache"
txDumpPath = "D:\Emulators\mupen64plus\texture_dump"
fontName = "arial.ttf"
fontSize = 18
fontColor = "B5E61D"
ForceGammaCorrection = False
GammaCorrectionLevel = 2
ShowFPS = False
ShowVIS = False
ShowPercent = False
ShowInternalResolution = False
ShowRenderingResolution = False
CountersPos = 8

Do note that the plugin can access the cache folder just fine, as it generated the file Banjo-Kazooie_MEMORYCACHE.htc there. But putting Banjo-Kazooie_HIRESTEXTURES.htc there changes nothing.

**Comment by gonetz** [comment]
Textures in texture cache created from texture pack. There are various options, which may alter resulted textures.  "txFullAlphaChannel" is one of these options. Plugins checks current plugin settings and if these settings differs from settings used to create the cache, plugin rebuilds the cache. Thus, if you created .htc file with Project64 and copied it to mupen64plus cache folder, you must set the same hires-tex related settings, as in Project64 to load that cache. If you have different settings, you must have texture pack with .png textures to rebuild the cache.

Unfortunately, some texture packs distributed only as .htc file, without texture pack itself.
If cache file does not correspond to plugin's settings, plugin tries to update texture cache, fails to find texture pack and does not load texture cache at all.
I added a condition: if texture pack for current game does not exists, load cache file regardless of plugin settings. Thus, now you only need to set txHiresEnable option and have .htc file in proper place.

**Comment by PisoMojado** [comment]
Hi, maybe I did something wrong, or that I misunderstand this issue, but I just wanted to offer feedback: I also can not load hires textures from my hires_textures directory.

I built the latest code including 5578519, then I put the texture file, ZELDA MAJORA'S MASK_HIRESTEXTURE.htc, in my hires_textures directory, but it did not load.

Solution for me: I then copied the texture file to my cache directory, and the textures loaded.
